### PR TITLE
New repo action: send comment on stale issues

### PR DESCRIPTION
**Pull request by WorldLanguages**
Messages from me on Discord:
> Is there a way to get a reminder if a specific GitHub issue hasn't had discussion for a specific period of time?
I sometimes expect someone to reply so I can remember about the issue, but if no one does, the issue falls into oblivion

This new GitHub action will not close any issues/PRs automatically.
It will not affect any issues/PRs, unless they have the `status: awaiting answer/followup` label. Only in issues/PRs that have that label, the bot will send a message after 7 days of inactivity.

**Example imaginary situation where I would use this:**
1. I comment on the issue/PR
image
2. I add the `status: awaiting answer/followup` label to the issue/PR.
3. In case mxmou (example) doesn't reply for 7 days, the bot will post a new comment. At that point, I can ping mxmou again, or merge the PR.
If mxmou replies in time, I simply remove the label.

The goal here is to make it harder for us to forget about issues/PRs in situations where we were expecting a reply or follow-up from someone.

### CODE REVIEW

File: .github/workflows/stale-issues.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    steps:
+      - uses: actions/stale@v6
+        with:
+          stale-issue-message: 'Ping! There has been no activity for 7 days.'
+          only-label: 'status: awaiting answer/followup'
+          days-before-issue-stale: 7
+          days-before-close: -1
+          stale-issue-label: ''
+          operations-per-run: 5
+          days-before-pr-stale: -1
```
Comment (WorldLanguages) [inline .github/workflows/stale-issues.yml:19]: TODO: PRs should go stale too.
At some point I thought this would make no sense in PRs, but I was wrong.

Comment (RedGuy12) [reply .github/workflows/stale-issues.yml:19]:
```suggest
          stale-message: 'Ping! There has been no activity for 7 days.'
          only-label: 'status: awaiting answer/followup'
          days-before-stale: 7
          days-before-close: -1
          stale-label: ''
          operations-per-run: 5
```
iirc

Comment (RedGuy12) [inline .github/workflows/stale-issues.yml:19]:
```suggestion
          stale-message: 'Ping! There has been no activity for 7 days.'
          only-label: 'status: awaiting answer/followup'
          days-before-stale: 7
          days-before-close: -1
          stale-label: ''
          operations-per-run: 5
```
iirc

Comment (WorldLanguages) [reply .github/workflows/stale-issues.yml]: Done

**Comment by Hans5958** [comment]
Maybe `status: awaiting response`?

**Comment by WorldLanguages** [comment]
> Maybe `status: awaiting response`?

That could work, but lately I noticed this tag is useful even in cases where there isn't a specific person that is expected to reply. "Followup" may indicate that the same contributor that added the tag expects to post soon. I just did that on issue 5311

**Comment by WorldLanguages** [comment]
Could anyone quickly review my settings file, to make sure everything is alright? And most importantly, verify I haven't made a whoopsie where the bot would close hundreds of issues. Theoretically, `operations-per-run: 5` should avoid that from ever happening, even if the configuration is incorrect.

**Comment by WorldLanguages** [comment]
Asking @RedGuy12 @apple502j @Hans5958 for quick approval?

**Comment by Hans5958** [comment]
I have doubts on you setting `operations-per-run: 5`. It may be a problem where it just not doing enough work. I think the default `30` should just be kept.

File: .github/workflows/stale-issues.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+          stale-pr-message: 'Ping! There has been no activity for 7 days.'
+          only-label: 'status: awaiting answer/followup'
+          days-before-stale: 7
+          days-before-close: -1
+          stale-issue-label: ''
+          stale-pr-label: ''
+          operations-per-run: 5
+    permissions:
+      issues: write
+      pull-requests: write
```
Comment (Hans5958) [inline .github/workflows/stale-issues.yml:23]: Super nitpick: Would it be possible to use the bot for this?

```suggestion
    runs-on: ubuntu-latest
    if: github.repository_owner == 'ScratchAddons'
    steps:
      - name: Generate token
        uses: tibdex/github-app-token@v1
        id: generate-token
        with:
          app_id: ${{ secrets.BOT_APP_ID }}
          private_key: ${{ secrets.BOT_PRIVATE_KEY }}
      - name: Execute Action
        uses: actions/stale@v6
        with:
          stale-issue-message: 'Ping! There has been no activity for 7 days.'
          stale-pr-message: 'Ping! There has been no activity for 7 days.'
          only-label: 'status: awaiting answer/followup'
          days-before-stale: 7
          days-before-close: -1
          stale-issue-label: ''
          stale-pr-label: ''
          operations-per-run: 5
          repo-token: ${{ steps.generate-token.outputs.token }}
```

File: .github/workflows/stale-issues.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+name: Send comment on stale issues
+on:
+  schedule:
+    - cron: '0 0 * * *'
+
+jobs:
+  stale:
```
Comment (Hans5958) [inline .github/workflows/stale-issues.yml:7]: Another super nitpick:

```suggestion
  stale:
    name: Stale
```

**Comment by WorldLanguages** [comment]
> I have doubts on you setting `operations-per-run: 5`. It may be a problem where it just not doing enough work. I think the default `30` should just be kept.

Okay, I'll change it to 30.

**Review by Hans5958 — APPROVED** [review]
We'll adjust the values if we got problems later.

**Pull request merged**